src/main/java/com/nvidia/spark/rapids/jni/JSONUtils.java

```diff
@@ -94,25 +100,41 @@ public static ColumnVector[] getJsonObjectMultiplePaths(ColumnVector input,
         indexes[pathOffsets[i] + j] = current.index;
       }
     }
-    long[] ptrs = getJsonObjectMultiplePaths(input.getNativeView(), typeNums,
-        names, indexes, pathOffsets);
-    ColumnVector[] ret = new ColumnVector[ptrs.length];
+    long[] ptrs = createGpuJSONPaths(typeNums, names, indexes, pathOffsets);
+    GpuJSONPath[] ret = new GpuJSONPath[ptrs.length];
     for (int i = 0; i < ptrs.length; i++) {
-      ret[i] = new ColumnVector(ptrs[i]);
+      ret[i] = new GpuJSONPath(ptrs[i]);
+    }
+    return ret;
+  }
+
+  public static ColumnVector getJsonObject(ColumnVector input, GpuJSONPath path) {
```

Comment:
nit: To avoid the merge problem, could we keep the old APIs also (for backwards compatibility) and just have something like.

```
GPUJSONPath[] gpuPaths = createGpuJSONPaths(Arrays.asList(path));
try {
  return getJsonObject(input, gpuPaths[0]);
} finally {
  for (GPUJSONPath p: gpuPaths) {
    if (p != nul) {
      p.close();
    }
  }
}
```

Then something similar for the multi case too.